src/common/iso_8601.cc

```diff
@@ -199,4 +199,54 @@ string to_iso_8601(const real_time t,
   out << '.' << setw(9) << nsec << 'Z';
   return out.str();
 }
+
+string to_iso_8601_no_separators(const real_time t,
+                                 const iso_8601_format f) noexcept {
+  ceph_assert(f >= iso_8601_format::Y &&
```

Comment:
should we be calling ceph_assert() here?  is this intended to be a compile-time check?